packages/c-live-region/src/use-live-region.ts

```diff
@@ -0,0 +1,14 @@
+import { reactive, watchEffect } from "vue"
+import { LiveRegion, LiveRegionOptions } from "./live-region"
+
+/**
+ * Creates a hidden live region with dynamic content based on triggered events
+ * to be read out by the screen reader on change of the content.
+ */
+export function useLiveRegion(options?: LiveRegionOptions) {
+  const liveRegion = reactive(() => new LiveRegion(options))
+
+  watchEffect((cleanup) => cleanup(() => liveRegion().destroy()))
+
+  return liveRegion()
+}
```

Comment:
After thinking about it, seem better to have a `CLiveRegion` component instead of a hook imo. 

All the options passed can basically be props and the rest of the code would be much simpler and more aligned with the Vue philosophy.
It will also allow to remove some of the code. I think that eventhough we need to be as close to the React package, there are some things that need to be done differently due to the different philosophy between React and Vue.

What is your opinion about this @codebender828 , @TylerAPfledderer  ?

Reply:
@Shyrro @codebender828 for context and my understanding of `LiveRegion`, it is a class because the hook triggers an instance which renders an element out of view of the visual user; a hidden portal of sorts specific to the use of a acreen reader. Then you pull methods (events) from the instance through the hook, to apply to what triggers are used in a given component.

The developer can certainly access the other data to make their own reason, but it did not seem to be intended by default for someone to have to make an element just for a screen reader to readout a simple message on change from some event.

I would not mind converting to a component, with composables to house the props that can be exported separately, but then the question would be how this new component is to be implemented in the environment.

I'm primarily thinking about how the `speak` method is currently executed versus an alternative approach.

Reply:
Tbh if you want to access methods of the components or other data you can expose them through the `expose` method of your component. They'll then be available through **template refs**

And for the fact the the component is outside the screen it doesn't really matter if you wrap it in a `Teleport` or `CPortal`

It just feels the "Vue way" so to say. Also, when manipulating the DOM directly through `document.createElement` etc, basically nothing is tracked so you're instance doesn't have any reactivity whatsoever.

I might be wrong about this since this is a special hook, but it just feels more natural in Vue to have in this case a **logical component** instead of a hook.

Reply:
Understood! I'll take a look at `expose`.

I'll play around with it, because you could probably argue the same case against direct DOM manipulation in React here with the reactivity, so maybe I can ask one of the React folks on the reason for this approach. 🤔

Better for me to know at least so I can get a better idea on differences! 😄

Reply:
Let me know if u need any more clarification on what i'm saying or any more explanation on this approach.

We can even discuss it in a call, might be easier ☺️